### PR TITLE
Add label cluster_name to alloy_config_hash and some cluster_ metrics

### LINKED ISSUE #1364: Use cluster name in alloy metrics
### Request

When `--cluster.name` is specified, use that in relevant metrics, like `alloy_config_hash`

### Use case

This will help differentiate between configs from the same alloy cluster or different alloy clusters.

### PR DESCRIPTION
**Pull request by wildum**
#### PR Description

- I added the label "cluster_name" to the metric alloy_config_hash and removed redundant metrics logic in config_metrics.
- I bumped the ckit version to also have the "cluster_name" label on relevant cluster_ metrics
- I updated the clustering alerts to take the new cluster_name label into account  

#### Which issue(s) this PR fixes

Fixes #1364

#### PR Checklist

- [x] CHANGELOG.md updated
- [ ] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

**Review by mattdurham — APPROVED** [review]
LGTM

File: operations/alloy-mixin/alerts/clustering.libsonnet [hunk 1]
```diff
@@ -9,7 +9,7 @@ local alert = import './utils/alert.jsonnet';
         alert.newRule(
           'ClusterNotConverging',
           if enableK8sCluster then 
-            'stddev by (cluster, namespace, job) (sum without (state) (cluster_node_peers)) != 0' 
+            'stddev by (cluster, namespace, job, cluster_name) (sum without (state) (cluster_node_peers)) != 0'
```
Comment (gaantunes) [inline operations/alloy-mixin/alerts/clustering.libsonnet:12]: We should name this new label as `alloy_cluster`, to clearly differentiate from the k8s `cluster` label. This is also the standard we have been using in other integrations, like `kafka_cluster`, `mongodb_cluster`, etc.

Comment (wildum) [reply operations/alloy-mixin/alerts/clustering.libsonnet:12]: the problem is that the cluster metrics are created by ckit library which is external to Alloy (https://github.com/grafana/ckit)

File: internal/static/config/instrumentation/config_metrics.go [hunk 2]
```diff
@@ -20,17 +20,20 @@ type configMetrics struct {
...
-func newConfigMetrics() *configMetrics {
+func newConfigMetrics(clusterName string) *configMetrics {
 	var m configMetrics
 
 	m.configHash = promauto.NewGaugeVec(
 		prometheus.GaugeOpts{
 			Name: "alloy_config_hash",
 			Help: "Hash of the currently active config file.",
+			ConstLabels: prometheus.Labels{
+				"cluster_name": clusterName,
```
Comment (gaantunes) [inline internal/static/config/instrumentation/config_metrics.go:35]: Isn't the name defined here?

File: operations/alloy-mixin/alerts/clustering.libsonnet (hunk 1 above)
Comment (gaantunes) [reply operations/alloy-mixin/alerts/clustering.libsonnet:12]: Ok, got it. Lets move on.

**Review by gaantunes — APPROVED** [review]
LGTM.

**Comment by gaantunes** [comment]
This is also fixing https://github.com/grafana/alloy/issues/1365

**Pull request merged**